src/main/java/frc/robot/commands/autonomousCommands/AutonomousShooterCommand.java

```diff
@@ -49,27 +53,27 @@ public void initialize(){
     public void execute(){
         shooterSpeedController.setLaunchSpeed(ShooterMath.determineLaunchSpeed(distanceToGoal + INNER_DISTANCE_FROM_TARGET));
 
-        for(int ballsShot = 0 ; ballsShot < 3 ; ballsShot ++){
-         // Make sure the shooter is at speed before loading a power cell
-         if (shooterSpeedController.isAtSpeed()) {
-             System.out.println("Shot Balls" + ballsShot);
-            // turn on the indexer and hopper
-            if(hasBall){
-                this.indexerController.intake();
-                this.hopperController.stop();
+        while(ballsShot < 3){
```

Comment:
You don't need a while loop here at all.  isFinished will detect when ballsShot is 3 and end this command.  In fact, we should stay away from while loops in general.  They have specific uses, but it's too easy to get stuck in one.